src/snowcli/cli/registry/manager.py

```diff
@@ -1,27 +1,101 @@
-from snowcli.cli.common.flags import ConnectionOption
-from snowcli.snow_connector import connect_to_snowflake
-
-
-def get_token(
-    environment: str = ConnectionOption,
-):
-    """
-    Get token to authenticate with registry.
-    """
-    conn = connect_to_snowflake(
-        connection_name=environment,
-        # to support registry login
-        session_parameters={"PYTHON_CONNECTOR_QUERY_RESULT_FORMAT": "json"},
-    )
-
-    # disable session deletion
-    conn._all_async_queries_finished = lambda: False
-    if conn._rest is None:
-        raise Exception("error in connection object")
-    # obtain and create the token
-    token_data = conn._rest._token_request("ISSUE")
-
-    return {
-        "token": token_data["data"]["sessionToken"],
-        "expires_in": token_data["data"]["validityInSecondsST"],
-    }
+import json
+import requests
+import sys
+import base64
+from urllib.parse import urlparse
+
+from snowcli.cli.common.sql_execution import SqlExecutionMixin
+from snowflake.connector.cursor import SnowflakeCursor
+
+
+class RegistryManager(SqlExecutionMixin):
+    def get_database(self):
+        return self._conn.database
+
+    def get_schema(self):
+        return self._conn.schema
+
+    def get_role(self):
+        return self._conn.role
+
+    def get_token(self):
+        """
+        Get token to authenticate with registry.
+        """
+        self._execute_query(
+            "alter session set PYTHON_CONNECTOR_QUERY_RESULT_FORMAT = 'json'"
+        ).fetchall()
```

Comment:
What's the purpose of it?

Reply:
I'm not quite sure, but this was in the original code for these commands.  I've tried without this option and the commands fail.  I can't see any actual difference in the tokens and I can't see any obvious reason that this is required.  However, I'm a bit short on time.  Think it's OK for us to revisit this one at a later date and leave it there?